ContextDesktop/src/main/java/com/github/migi_1/Context/model/Environment.java

```diff
@@ -119,11 +119,21 @@ public void removeEntity(Entity entity) {
 	 */
 	private void moveMovables() {
 		for (IMovable movable : movables) {
-			movable.move(movable.getMoveBehaviour().getMoveVector());
+			movable.move(addDifficulty(movable.getMoveBehaviour().getMoveVector()));
 		}
 	}
 
 	/**
+	 * Add difficulty to the moveVector of the movables.
+	 * @param movableVector
+	 */
+	private Vector3f addDifficulty(Vector3f movableVector) {
+	    return new Vector3f(movableVector.x * difficultyFactor,
+	                        movableVector.y * difficultyFactor,
+	                        movableVector.z * difficultyFactor);
+	}
```

Comment:
Doesn't Vector3f have a method so that you can easily multiply with a certain value?